packages/core/lib/runner.js

```diff
@@ -223,6 +224,16 @@ function run(script, ee, options, runState, contextVars) {
 
         const totals = SSMS.pack(intermediates);
 
+        runState.engines.map((engine) => {
+          if (engine.cleanup) {
+            try {
+              engine.cleanup().then(() => debug('engine unloaded:', engine.__name)) ;
```

Comment:
This should be refactored to use async/await. Otherwise there's no guarantee that a given `cleanup()` handler finishes before the CLI exits, or that they complete in the order they're being called.